app/models/miq_smart_proxy_worker/runner.rb

```diff
@@ -1,3 +1,73 @@
 class MiqSmartProxyWorker::Runner < MiqQueueWorkerBase::Runner
   self.delay_startup_for_vim_broker = true # NOTE: For smartproxy role
+
+  def do_before_work_loop
+    @tid = start_heartbeat_thread
+  end
+
+  def before_exit(message, _exit_code)
+    @exit_requested = true
+    #
+    # Stop the Heartbeat Thread
+    #
+    safe_log("#{message} Stopping Heartbeat Thread.")
+
+    #
+    # Wait for the Heartbeat Thread to stop
+    #
+    unless @tid.nil?
```

Comment:
Prefer the positive case: `if @tid`

Reply:
That's fine.  Please note that a lot of this was glommed from the monitor_thread in the event_catcher.  I note that this comment and that at line 61 below were direct copies from there.  I will change it here but it still lives there.